### PR TITLE
Support reserve cache without size information

### PR DESCRIPTION
**Pull request by velioglu**
It was assumed that reserve cache sends the size information, though it is not valid for all cases (docker layer cache is an example). In that case, reserve cache returns presigned urls for the biggest cache file size. Size of the entry is updated during commit cache as it is always sent with that request.

### CODE REVIEW

File: routes/runtime/github.rb [hunk 1]
```diff
@@ -82,7 +82,7 @@ class CloverRuntime
           fail CloverError.new(409, "AlreadyExists", "A cache entry for #{scope} scope already exists with #{key} key and #{version} version.")
         end
 
-        if size > GithubRepository::CACHE_SIZE_LIMIT
+        if size&.> GithubRepository::CACHE_SIZE_LIMIT
```
Comment (enescakir) [inline routes/runtime/github.rb:85]: nit: It's very hard to read

```suggestion
        if size && size > GithubRepository::CACHE_SIZE_LIMIT
```

File: routes/runtime/github.rb [hunk 2]
```diff
@@ -113,7 +118,12 @@ class CloverRuntime
         fail CloverError.new(400, "InvalidRequest", "Wrong parameters") if etags.nil? || etags.empty? || upload_id.nil? || size == 0
 
         entry = GithubCacheEntry[repository_id: repository.id, upload_id: upload_id, committed_at: nil]
-        fail CloverError.new(204, "NotFound", "No cache entry") if entry.nil? || entry.size != size
+        fail CloverError.new(204, "NotFound", "No cache entry") if entry.nil?
```
Comment (enescakir) [inline routes/runtime/github.rb:121]: Should we continue to check size if it's provided?
```suggestion
        fail CloverError.new(204, "NotFound", "No cache entry") if entry.nil? || (entry.size && entry.size != size)
```

File: routes/runtime/github.rb [hunk 3]
```diff
@@ -113,7 +118,12 @@ class CloverRuntime
...
+
+        # If the size can not be set with reserveCache, we set it here.
+        if entry.size.nil?
+          entry.update(size: size)
+        end
```
Comment (enescakir) [inline routes/runtime/github.rb:126]: nit: 
```suggestion
        entry.update(size: size) if entry.size.nil?
```

Comment (enescakir) [reply routes/runtime/github.rb:126]: btw we already send an update query to database in the next lines: `entry.update(committed_at: Time.now)`. Can you combine them?

Comment (velioglu) [reply routes/runtime/github.rb:126]: I prefer to keep it explicit after checking whether the size is nil.

Comment (enescakir) [reply routes/runtime/github.rb]: I agree with to keep it explicit. You can combine them like 
```ruby
updates = {committed_at: Time.now}
updates[:size] = size if entry.size.nil?
entry.update(updates)
```

**Pull request merged**